hayeon/programmers-0/옹알이.js

```diff
@@ -0,0 +1,19 @@
+function solution(babbling) {
+  let answer = 0;
+  for (let i = 0; i < babbling.length; i++) {
+    const target = babbling[i];
+    if (!target.match(/(ayaaya|yeye|woowoo|mama)/g)) {
+      if (target.replace(/(aya|ye|woo|ma)/g, "").length === 0) {
+        answer++;
+      }
+    }
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(babbling) {
  const regex = /(aya|ye|woo|ma)/g; // 정규표현식을 따로 변수로 두기
  let answer = 0;
  
  // 줄바꿈해서 변수 선언부와 for문 분리해서 가독성 챙기기1
  
  for (let i = 0; i < babbling.length; i++) {
    const replacedBabbling = babbling[i].replace(regex, ""); 
    // 변수로 둔 정규표현식을 이용하니 깔끔
    
    if (!replacedBabbling) answer++;
  }
    
  return answer;
}
```
- 음 코드 자세히 테스트 해보셨는지는 모르겠는데 if문 match문은 현재 풀이에서 필요 없습니다~ 
- 그리고 if문에안에 !target.match(/(ayaaya|yeye|woowoo|mama)/g), target.replace(/(aya|ye|woo|ma)/g, "").length === 0 둘다 `너무!` 가독성에 안좋습니다. 변수로 빼는 것을 추천드립니다. 개선 코드에서는 match문은 필요없어 제거했고 replace문은 replacedBabbling을 선언해줬습니다.
- 또한, replace를 `""` 로 변화시켜주기 때문에 길이는 체크안하셔도 되고, `""`면 false로 평가되기 때문에 `!replacedBabbling`일 경우를 체크해주시면 됩니다.
- 또한 예전에도 말씀드렸지만 문단별로 줄바꿈을 해서 코드를 가독성있게 만들어주세요 (변수 선언부, for문) 이런 기본적인 컨벤션은 제가 언급안드려도 하연님이 계속 신경쓰셔야 됩니다..🥲
- 아래와 같이 reduce를 통한 풀이도 가능합니다.
```js
function solution(babbling) {
  const regex = /aya|ye|woo|ma/g;

  return babbling.reduce((acc, cur) => {
    const replacedBabbling = cur.replace(regex, "");

    return !replacedBabbling ? acc + 1 : acc;
  }, 0);
}
```

Reply:
넵넵 ㅠㅠ 시험때문인건 너무 핑계같지만,, 제가 너무 정신없이 했나봐요 죄송합니다ㅜㅜ 다음주 코드는 더욱더 신경써서 쓰겠습니다! 😭